File: dotnet/azure/Mcma.Azure.BlobStorage/Proxies/BlobStorageFolderProxy.cs
using System.IO;
using System.Text;
using System.Threading.Tasks;
using Microsoft.Azure.Storage.Blob;

namespace Mcma.Azure.BlobStorage.Proxies
{
    public class BlobStorageFolderProxy : BlobStorageProxy<BlobStorageFolderLocator>
    {
        internal BlobStorageFolderProxy(BlobStorageFolderLocator locator, CloudBlobClient client)
            : base(locator, client)
        {
        }

        public async Task<BlobStorageFileLocator> PutAsync(string fileName, Stream readFrom)
        {
            var fileLocator = Locator.FileLocator(fileName);
            var blobRef = Container.GetBlockBlobReference(fileLocator.FilePath);
            await blobRef.UploadFromStreamAsync(readFrom);
            return fileLocator;
        }

        public async Task<BlobStorageFileLocator> PutAsTextAsync(string fileName, string content)
            => await PutAsync(fileName, new MemoryStream(Encoding.UTF8.GetBytes(content)));
    }
}

File: nodejs/azure/blob-storage/lib/proxies/blob-storage-folder-proxy.ts
import { Readable } from "stream";
import { BlobServiceClient } from "@azure/storage-blob";

import { BlobStorageFolderLocator } from "../blob-storage-folder-locator";
import { BlobStorageFileLocator } from "../blob-storage-file-locator";
import { getFileLocator } from "../helpers";
import { BlobStorageProxy } from "./blob-storage-proxy";

export class BlobStorageFolderProxy extends BlobStorageProxy<BlobStorageFolderLocator> {
    constructor(locator: BlobStorageFolderLocator, client: BlobServiceClient) {
        super(locator, client);
    }

    async put(fileName: string, readFrom: Readable): Promise<BlobStorageFileLocator> {
        const fileLocator = getFileLocator(this.locator, fileName);
        const blob = this.container.getBlockBlobClient(fileLocator.filePath);
        await blob.uploadStream(readFrom);
        return fileLocator;
    }

    async putAsText(fileName: string, content: string): Promise<BlobStorageFileLocator> {
        const fileLocator = getFileLocator(this.locator, fileName);
        const blob = this.container.getBlockBlobClient(fileLocator.filePath);
        await blob.upload(content, content.length);
        return fileLocator;
    }
}
